src/outbound.rs

```diff
@@ -263,6 +263,66 @@ impl Outbound {
         })
     }
 
+    pub(crate) fn drain(
+        &self,
+        sids: Vec<usize>,
+        pong_rx: crossbeam_channel::Receiver<bool>,
+    ) -> io::Result<()> {
+        // take lock, which will be held for the duration
+        // of this operation
+        let mut writer = self.writer.lock();
+
+        if writer.is_disconnected() {
```

Comment:
Should this error or just return?

Reply:
well, if they are calling drain for some assurance that they are receiving things, then we may not want to give them that assurance if the system is in a state of decreased availability. but I don't know what the expectations of users are. If people call drain without any hope that the server is in communication then a return is OK.

Reply:
If they are disconnected there is nothing to drain ;)

Reply:
what if they just published some things into a topic that they are also subscribing to, as a way of seeing round-trips? shouldn't they get an error if they're calling drain as a kind of "barrier" operation the way that they would call "flush"?